core/src/main/scala/kafka/log/Log.scala

```diff
@@ -753,6 +753,7 @@ class Log(@volatile var dir: File,
     debug("Closing log")
     lock synchronized {
       checkIfMemoryMappedBufferClosed()
+      producerExpireCheck.cancel(true)
```

Comment:
you should also wait until the executor has finished running, with something like `ExecutorService#awaitTermination`

Reply:
I don't think you can do that. The executor is shared with other components.

Reply:
So should I keep as it or change it?

Reply:
@ijuma: Good point.  We don't want to wait for the executor to terminate here.